crates/evm/traces/src/debug/sources.rs

```diff
@@ -212,6 +212,12 @@ impl ContractSources {
         Ok(())
     }
 
+    /// Merges given contract sources.
+    pub fn merge(&mut self, sources: Self) {
+        self.sources_by_id.extend(sources.sources_by_id);
+        self.artifacts_by_name.extend(sources.artifacts_by_name);
```

Comment:
```suggestion
        for (name, artifacts) in sources.artifacts_by_name {
            self.artifacts_by_name.entry(name).or_default().extend(artifacts);
        }
```

we might have duplicating name in different source sets,